app/src/main/res/values/strings.xml

```diff
@@ -75,6 +76,9 @@
   <string name="login_success">Login success!</string>
   <string name="login_failed">Login failed!</string>
   <string name="upload_failed">File not found. Please try another file.</string>
+  <string name="retry_limit_reached">Maximum retry limit reached! Please cancel the upload and try again</string>
+  <string name="unrestricted_battery_mode">Turn battery optimisations off</string>
```

Comment:
I would add a ? at the end to make it friendlier. Also, "optimization" (singular not plural) is less freightening and reflects the wording in the OS settings.

Reply:
Oops, I'll use 'z' instead of 's' everywhere.

Reply:
Ah sorry, I only meant about the singular. Either optimization or optimisation is fine for me 😊